homeassistant/components/bouncie/config_flow.py

```diff
@@ -0,0 +1,96 @@
+"""Config flow for bouncie integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from bounciepy import AsyncRESTAPIClient
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+# DONE adjust the data schema to the data that you need
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_CLIENT_ID): str,
+        vol.Required(
+            CONF_CLIENT_SECRET,
+        ): str,
+        vol.Required("redirect_uri"): str,
+        vol.Required("code"): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    # If your PyPI package is not built with async, pass your methods
+    # to the executor:
+    # await hass.async_add_executor_job(
+    #     your_validate_func, data["username"], data["password"]
+    # )
+
+    bouncie_client = AsyncRESTAPIClient(
+        client_id=data["client_id"],
+        client_secret=data["client_secret"],
+        redirect_url=data["redirect_uri"],
+        auth_code=data["code"],
+    )
+    result = await bouncie_client.get_access_token()
+    if not result:
+        raise InvalidAuth
+    data["access_token"] = bouncie_client.access_token
+    # Return info that you want to store in the config entry.
+    return data
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for bouncie."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            return self.async_create_entry(title="bouncie", data=info)
```

Comment:
```suggestion
            return self.async_create_entry(title="Bouncie", data=info)
```
Title with the name of the user if possible